File: src/main/java/no/nav/nav/sbl/dialogarena/modiabrukerdialog/api/service/gsak/SakerService.java
package no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.service.gsak;

import no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.domain.gsak.Sak;
import no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.domain.gsak.Saker;
import no.nav.nav.sbl.dialogarena.modiabrukerdialog.api.exceptions.JournalforingFeilet;

import java.util.List;

public interface SakerService {
    Saker hentSaker(String fnr);

    List<Sak> hentListeAvSaker(String fnr);

    List<Sak> hentRelevanteSaker(String fnr);

    void knyttBehandlingskjedeTilSak(String fnr, String behandlingskjede, Sak sak) throws JournalforingFeilet;

    void knyttBehandlingskjedeTilSak(String fnr, String behandlingskjede, Sak sak, String enhet) throws JournalforingFeilet;
}

File: web/src/main/java/no/nav/modiapersonoversikt/legacy/api/service/saker/SakerService.kt
package no.nav.modiapersonoversikt.legacy.api.service.saker

import no.nav.modiapersonoversikt.legacy.api.domain.saker.Sak

interface SakerService {
    class Resultat(val saker: MutableList<Sak> = mutableListOf(), val feiledeSystemer: MutableList<String> = mutableListOf())

    fun hentSaker(fnr: String): Resultat

    fun hentSakSaker(fnr: String): Resultat
}
